ModelViewer/ModelViewer/ViewerGraphicsWindow.cpp

```diff
@@ -56,6 +57,74 @@ bool ViewerGraphicsWindow::loadModel(QString filepath) {
     return m_currentModel.m_isValid;
 }
 
+void ViewerGraphicsWindow::mousePressEvent(QMouseEvent* event)
+{
+    // Set class vars
+    if (event->buttons() & Qt::LeftButton) {
+        leftMousePressed = true;
+    }
+
+    if (event->buttons() & Qt::RightButton) {
+        rightMousePressed = true;
+    }
+
+    // Make sure that these are set before the mouseMoveEvent triggers
+    lastX = event->x();
+    lastY = event->y();
+    // Call the partent class 
+    QWindow::mouseReleaseEvent(event);
+}
+
+void ViewerGraphicsWindow::mouseReleaseEvent(QMouseEvent* event)
+{
+    // Set class vars
+    if (event->buttons() & Qt::LeftButton) {
+        leftMousePressed = false;
+    }
+
+    if (event->buttons() & Qt::RightButton) {
+        rightMousePressed = false;
+    }
+    // Call the partent class
+    QWindow::mouseReleaseEvent(event);
+}
+
+void ViewerGraphicsWindow::mouseMoveEvent(QMouseEvent* event)
+{
+    float deltaX = lastX - event->x();
+    float deltaY = lastY - event->y();
+
+    // RMB: Rotate off of x y movement
+    if (event->buttons() & Qt::RightButton) {
+        // TODO: This could be better. If we keep track of the normal of the model
+        // we could make sure that translateing in x & y won't pitch the object.
+        sceneMatrix.rotate(-deltaX * xRotateSensitivity, 0, 1, 0);
+        sceneMatrix.rotate(-deltaY * yRotateSensitivity, 1, 0, 0);
+    }
+
+    // MMB: Pan off of x y movement
+    if (event->buttons() & Qt::LeftButton) {
+        viewportX += -deltaX * viewportXSensitivity;
```

Comment:
Hmmm, never seen pan implemented by changing the viewport...  Seems interesting.  We can merge this for now, but maybe add a comment like `TODO: Change to QMatrix4x4.translate()`.  I think that would be a cleaner way to do it...

Reply:
I recommend keeping a separate matrix for translation, and rotation, and view (matrix.perspective).  Then, in the render() function, compute the overall transformation by multiplying these together each frame

Reply:
Created #32